optuna/study/study.py

```diff
@@ -1169,10 +1170,9 @@ def objective(trial):
         sampler = samplers.NSGAIISampler()
 
     study_name = storage.get_study_name_from_id(study_id)
+    storage.set_study_directions(study_id, direction_objects)
```

Comment:
Almost LGTM, but I'm not confident about the consequence of moving this line before creating `Study`. Could you explain the motivation behind this?

Reply:
In order to apply [this change](https://github.com/optuna/optuna/pull/4146#issuecomment-1308045371), we must ensure that directions must be set before instantiating `Study` object. Is my answer what you need?

Reply:
I see, thanks!